firestore/integration_test_internal/src/aggregate_query_snapshot_test.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+
+using AggregateQuerySnapshotTest = FirestoreIntegrationTest;
+
+std::size_t AggregateQuerySnapshotHash(const AggregateQuerySnapshot& snapshot) {
+  return snapshot.Hash();
+}
+
+TEST_F(AggregateQuerySnapshotTest, Equality) {
```

Comment:
Please break up the Equality test into multiple test cases, to improve readability. Each test case should test one and only one behavior. Without studying this function in great detail it's nearly impossible to determine which cases are being covered. 

go/unit-testing-practices#behavior-testing-examples